Documentation/contributing/development/dev_setup.rst

```diff
@@ -318,7 +318,25 @@ Lets assume we want to add ``github.com/containernetworking/cni`` version ``v0.5
 For a first run, it can take a while as it will download all dependencies to
 your local cache but the remaining runs will be faster.
 
-Updating k8s is a special case, for that one needs to do:
+Updating k8s is a special case, for that one needs to update the client-go
+library first and then the remaining dependencies:
+
+#. Clone ``https://github.com/kubernetes/client-go``
+#. Add the new remote fork: ``git remote add origin-cilium https://github.com/cilium/client-go.git``
+#. Fetch all branches from the remote fork ``git fetch origin-cilium``
+#. Checkout to the release that we are updating ``git fetch origin --tags && git checkout kubernetes-1.19.0``
+#. Create a new branch ``git checkout -b kubernetes-1.19.0-with-hot-fix``
+#. Cherry pick the patches that are not merged upstream from the older branches,
+   there are currently 2 patches not merged upstream:
```

Comment:
> 2 patches not merged

This will date quickly unless we plan to maintain two specific patches long term. Can we make this more general to figure out the patches that must be applied?

I think in the case of Envoy patches, we explicitly maintain these as patch files in a directory in the repo and apply them on top. (Not required for this case, just a data point).

Reply:
@joestringer Due lack of cycles on my side this could have been 0. It has been 2 patches since the fork was created. I'll try to push for this in the next months.